spec/expression.dd

```diff
@@ -2748,7 +2748,8 @@ void foo()
         $(H5 $(LNAME2 is-type-equal, $(D is $(LPAREN)) $(I Type) $(D ==) $(I TypeSpecialization) $(D $(RPAREN))))
```

Comment:
I had always assumed that in `is(foo == keyword)` that foo was a symbol, not a type. 
```d
int x;
pragma(msg, is(x == float)); // false, also x is not a type.
```

Reply:
AFAIK only `== module` and `== package` are satisfied when the argument is not a type. That's why they break the concept of IsExpression testing if a type is valid.

@RazvanN7 Perhaps they would be a good candidate for the [Language Simplification project](https://github.com/orgs/dlang/projects/11/views/1).

Reply:
They really aren't, they have only one meaning, that meaning is obvious, and that form does not make sense to repurpose.

Reply:
Then at minimum the grammar saying Type should be changed just for the module and package forms, because they do not accept a type, unlike all the forms which require only a type.

Reply:
I suppose the grammar is technically OK, because Type includes QualifiedIdentifier which can accept ModuleFullyQualifiedName or PackageName, even though it's not actually a type.

Reply:
The source code actually states:
> if (e.tok2 == TOK.package_ || e.tok2 == TOK.module_) // These is() expressions are special because they can work on modules, not just types.

https://github.com/dlang/dmd/commit/9952fbb6c29#diff-f0a120d764c7fab59bd3210b19fe685d2f106e4a1b9a33266804ae99dd3bbd4eR5500

Special cases need strong justification to justify the cost of the complexity on the user.